lib/vault/api/auth.rb

```diff
@@ -186,6 +190,63 @@ def aws_ec2(role, pkcs7, nonce = nil)
       return secret
     end
 
+    # Authenticate via IAM EC2 method by providing a AWS CredentialProvider (either ECS, AssumeRole, etc.)
+    # If authentication is successful, the resulting token will be stored on the client and used
+    # for future requests.
+    #
+    # @example
+    #   Vault.auth.aws_ecs_iam("dev-role-iam", Aws::AssumeRoleCredentials.new, "https://sts.us-east-2.amazonaws.com", "vault.example.com") #=> #<Vault::Secret lease_id="">
+    #
+    # @param [String] role
+    # @param [CredentialProvider] credentials_provider
+    #   https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/CredentialProvider.html
+    # @param [String] sts_endpoint optional
+    #   https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html
+    # @param [String] iam_auth_header_value optional
+    #
+    # @return [Secret]
+    def aws_iam(role, credentials_provider, sts_endpoint = 'https://sts.amazonaws.com', iam_auth_header_value = IAM_SERVER_ID_HEADER)
+      require "aws-sigv4"
+      require "base64"
+
+      valid_sts_endpoint = %r{https:\/\/sts.?(.*).amazonaws.com}.match(sts_endpoint)
```

Comment:
FWIW, the STS endpoint for STS in the [China (Beijing) region (cn-north-1)](http://docs.amazonaws.cn/en_us/general/latest/gr/rande.html#cnnorth_region) is `sts.cn-north-1.amazonaws.com.cn`. This will still work because your match doesn't have a `$` at the end, but might be worth commenting so that future people don't add it and inadvertently break support for cn-north-1.

The [endpoint for GovCloud](https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-endpoints.html) is `sts.us-gov-west-1.amazonaws.com` so it should work exactly as intended.